src/SourceCode.Clay.Algorithms/TreeNode.cs

```diff
@@ -80,7 +89,7 @@ public override int GetHashCode()
         {
             var hashCode = -1781160927;
 
-            hashCode = hashCode * -1521134295 + EqualityComparer<T>.Default.GetHashCode(Node);
+            hashCode = hashCode * -1521134295 + _equalityComparer.GetHashCode(Node);
```

Comment:
`_equalityComparer` may be null (struct). Introduce a local `var equalityComparer = _equalityComparer ?? EqualityComparer<T>.Default`.